src/basilisp/contrib/bencode.lpy

```diff
@@ -0,0 +1,134 @@
+(ns basilisp.contrib.bencode
+  "A port of nbb's bencode implementation to basilisp.
+
+  https://github.com/babashka/nbb/blob/bca8b5017a06768eb35d02a2d6233ca9c6c2f692/src/nbb/impl/bencode.cljs")
+
+(defn- index-of [s c]
+  (let [i (.index s c)]
+    (if (< i 0) (throw (python/Exception "out of input")) i)))
+
+(defn- slice
+  ([buffer start]
+   (if (< (len buffer) start)
+     (throw (python/Exception "out of input"))
+     (.__getitem__ buffer (python/slice start nil))))
+  ([buffer start end]
+   (if (> end (len buffer))
+     (throw (python/Exception "out of input"))
+     (.__getitem__ buffer (python/slice start end)))))
```

Comment:
Kind of a semantic argument here but I'm wondering if something like an `EOFError` or even just `ValueError` would be a little bit more appropriate here.

Reply:
Good point, changed to EOFError.